rocAL/rocAL/source/api/rocal_api_data_loaders.cpp

```diff
@@ -272,6 +272,87 @@ rocalJpegFileSource(
     return output;
 }
 
+RocalImage  ROCAL_API_CALL
+rocalJpegExternalFileSource(
+        RocalContext p_context,
+        const char* source_path,
+        RocalImageColor rocal_color_format,
+        bool is_output,
+        bool shuffle,
+        bool loop,
+        RocalImageSizeEvaluationPolicy decode_size_policy,
+        unsigned max_width,
+        unsigned max_height,
+        RocalDecoderType dec_type,
+        RocalExtSourceMode external_source_mode)
+{
+    Image* output = nullptr;
+    auto context = static_cast<Context*>(p_context);
+    try
+    {
+        uint internal_shard_count = 1;
+        bool use_input_dimension = (decode_size_policy == ROCAL_USE_USER_GIVEN_SIZE) || (decode_size_policy == ROCAL_USE_USER_GIVEN_SIZE_RESTRICTED);
+        bool decoder_keep_original = (decode_size_policy == ROCAL_USE_USER_GIVEN_SIZE_RESTRICTED) || (decode_size_policy == ROCAL_USE_MAX_SIZE_RESTRICTED);
+        DecoderType decType = DecoderType::TURBO_JPEG; // default
+        if (dec_type == ROCAL_DECODER_OPENCV) decType = DecoderType::OPENCV_DEC;
+        if ((decode_size_policy == ROCAL_USE_MAX_SIZE) || (decode_size_policy == ROCAL_USE_MAX_SIZE_RESTRICTED))
+            THROW("use_max_size is not supported in external source reader");
+
+        // user need to specify this
+        if(max_width == 0 || max_height == 0)
+        {
+            THROW("Invalid input max width and height");
+        }
+        else
+        {
+            LOG("User input size " + TOSTR(max_width) + " x " + TOSTR(max_height))
+        }
+
+        auto [width, height] = std::make_tuple(max_width, max_height);
+
+        auto [color_format, num_of_planes] = convert_color_format(rocal_color_format);
+
+
+        INFO("Internal buffer size width = "+ TOSTR(width)+ " height = "+ TOSTR(height) + " depth = "+ TOSTR(num_of_planes))
+        context->set_internal_batch_size(context->user_batch_size());   // for external_source internal batch size has to be same as user
```

Comment:
Please change the comment to
// for external_source internal batch size has to be same as user batch size